thumbor/engines/pil.py

```diff
@@ -342,6 +352,30 @@ def convert_to_grayscale(self, update_image=True, with_alpha=True):
             self.image = image
         return image
 
+    def has_transparency(self):
+        has_transparency = 'A' in self.image.mode or 'transparency' in self.image.info
+        if has_transparency and numpy:
+            # If has transapency and have numpy
+            # check if any pixel in Alpha layer is 255(opaque)
+            img = self.image.convert('RGBA')
+            im = numpy.array(img)
+            alpha = im[:, :, -1]  # just alpha layer
+            has_transparency = alpha.min() < 255
+        return has_transparency
+
+    def can_auto_convert_png_to_jpg(self, *args, **kwargs):
+        can_convert = super(Engine, self).can_auto_convert_png_to_jpg(*args, **kwargs)
+
+        if not can_convert:
+            return can_convert
+
+        if not self.has_transparency():
```

Comment:
Maybe not include this check? If the png can be converted (no transparent pixels), do it. Otherwise, keep png.